tests/python/pants_test/engine/test_fs.py

```diff
@@ -508,6 +509,31 @@ def test_glob_match_warn_logging(self):
       single_warning = all_warnings[0]
       self.assertEqual("???", str(single_warning))
 
+  def test_files_content_from_bytes(self):
+    # TODO: Should you be allowed to set executable? (currently we always set executable=true)
+    # TODO: Should there be a way to represent empty directories?
+    # TODO: Test with some unicode, some non-unicode byte, and a directory with multiple files
+    files_content = FilesContent((
+      FileContent(
+        path=text_type("cats/roland"),
+        content=b"European Burmese",
+      ),
+      FileContent(
+        path=text_type("treats"),
+        content=b"catnip",
+      )
+    ))
+
+    snapshot = assert_single_element(self.scheduler.product_request(Snapshot, [files_content]))
+    self.assertEquals(("cats/roland", "treats"), snapshot.files)
+    self.assertEquals(
+      snapshot.directory_digest,
+      Digest(
+        fingerprint="1249edd274db4350ce7d295dc2f40768581197f0e13e74d0206a3c408f1c236f",
```

Comment:
For the sake of me testing things in the future, how did you figure out this value?

Reply:
We have this handy binary called `fs_util`! Run:
`build-support/bin/native/cargo run -p fs_util --manifest-path=src/rust/engine/Cargo.toml -- --help` for some helptext :) In particular, `build-support/bin/native/cargo run -p fs_util --manifest-path=src/rust/engine/Cargo.toml -- directory save --help`